### PR TITLE
[16.0][mig] project_deadline: Migration to 16.0

### PR DESCRIPTION
**Pull request by anusriNPS**
Migrating project_deadline to 16.0

Reopened PR: https://github.com/OCA/project/pull/1123

### CODE REVIEW

**Comment by anusriNPS** [comment]
@heliaktiv , I have reopened PR of migrating project_deadline to 16.0 and I would request you to review it

**Review by PicchiSeba — APPROVED** [review]
Code review: LGTM

**Comment by HekkiMelody** [comment]
Please squash administrative commits, as described here: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

File: project_deadline/tests/test_project_deadline.py
Comment (HekkiMelody) [inline project_deadline/tests/test_project_deadline.py]: **question**: why were these two lines changed to be the opposite of what they were before?

**Comment by anusriNPS** [comment]
I tried to squash the commit, but it resulted in merge conflict issue related to project_hr commits, so could not do it.

When git rebase -i origin/16.0 is executed, it lists other project related repos like project_hr and showing merge conflicts related to hr project during squash

image

Is there any way to get only project_deadline related commits alone, so that it can be squashed?

**Comment by HekkiMelody** [comment]
> I tried to squash the commit, but it resulted in merge conflict issue related to project_hr commits, so could not do it.
> 
> When git rebase -i origin/16.0 is executed, it lists other project related repos like project_hr and showing merge conflicts related to hr project during squash
> 
> image
> 
> Is there any way to get only project_deadline related commits alone, so that it can be squashed?

I suspect your repo may be outdated. Can you try `git fetch origin 16.0` and then try to rebase again?

File: project_deadline/tests/test_project_deadline.py
Comment (anusriNPS) [reply project_deadline/tests/test_project_deadline.py]: As part of Odoo 16.0 addons project view, date_start and date fields are included in the xml whose external id is "edit_project" which is displayed as "Planned Date". 

https://github.com/odoo/odoo/blob/16.0/addons/project/views/project_views.xml#L506C28-L511C119


Screenshot from 2024-10-22 16-45-36

 
 As this is available as part of Odoo addons project module, I am removing project_project related code changes which appends date_start and date field to the form view from project_deadline module.
 Also, the testcase related to it is also no longer valid for 16.0 version

File: project_deadline/views/project_project_view.xml
Comment (HekkiMelody) [inline project_deadline/views/project_project_view.xml]: **suggestion** (non-blocking): Instead of comparing two strings, we could turn `record.date.raw_value` into a Date object. See here for an example:

https://github.com/OCA/social/blob/16.0/mail_activity_board/views/mail_activity_view.xml#L186-L190

File: project_deadline/views/project_project_view.xml
Comment (HekkiMelody) [inline project_deadline/views/project_project_view.xml]: **issue**(non-blocking): This could be done more elegantly without the `t-if` and `t-else`, maybe with a `t-att-style`?

**Review by HekkiMelody — CHANGES_REQUESTED** [review]
Code review, mostly looks good to me, only a couple of suggestions.

I also think the changes to the translation files should be removed from the migration commit.

File: project_deadline/views/project_project_view.xml
Comment (anusriNPS) [reply project_deadline/views/project_project_view.xml]: Resolved

File: project_deadline/views/project_project_view.xml
Comment (anusriNPS) [reply project_deadline/views/project_project_view.xml]: resolved

File: project_deadline/views/project_project_view.xml
Comment (HekkiMelody) [inline project_deadline/views/project_project_view.xml]: **issue**:  Tested on runboat. We already have this info in the kanban.

image

(The date above is added by this module, the two dates below are in the base module)

Looking at the code, even the styling is the same. So we don't need this view at all.

**Review by HekkiMelody — COMMENTED** [review]
Code review, ok.

However, upon functional review, I wonder if this module is worth migrating at all. Since we don't need to add the core fields that this module would offer, all we're adding is two related fields on `project.task`.

At the very least, the readme should be updated to reflect the much more limited scope of this module, but maybe the module can be dropped entirely.

File: project_deadline/views/project_project_view.xml
Comment (anusriNPS) [reply project_deadline/views/project_project_view.xml]: Though I saw this information on the kanban view , since in 14.0 date was shown seperately near Partner_id I thought it should be shown in 16.0 even we have start and end date, hence added this.

Removed some of the code as those were not necessary in 16.0 however I misinterpreted this date information behaviour and added here

image

Comment (anusriNPS) [reply project_deadline/views/project_project_view.xml]: When project deadline is exceeded, in the base view date color is not changed to "Red" which is different from 14.0 view.

**Suggestion:** Instead of including new field which shows exceeded date in "Red", date already available can be shown in "Red" color.

image

Comment (anusriNPS) [reply project_deadline/views/project_project_view.xml]: Latest patchset:

image

When partner id is assigned, deadline surpassed will be indicated in "Red"

**Review by HekkiMelody — APPROVED** [review]
Code review, LGTM.

My doubts about whether this module is really useful remain, but I'll leave it up to a maintainer to decide.

**Comment by coleste** [comment]
Code review: LGTM

**Review by Daniel-Melian — COMMENTED** [review]
Functionally reviewed, and everything works correctly. However, I agree with @aleuffre  that this module may not be necessary.

**Comment by anusriNPS** [comment]
As this module is not necessary for migrating into 16.0, closing this MR

**Pull request closed without merging**
